packages/system/src/Location/ListView/LocationEditModal/LocationEditModal.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { FC, useState } from 'react';
+import {
+  ModalMode,
+  useDialog,
+  BasicTextInput,
+  Grid,
+  DialogButton,
+  useTranslation,
+  generateUUID,
+  ToggleButton,
+  InlineSpinner,
+} from '@openmsupply-client/common';
+import { Location } from '../../types';
+import { useLocationInsert, useNextLocation, useLocationUpdate } from '../api';
+interface LocationEditModalProps {
+  mode: ModalMode | null;
+  isOpen: boolean;
+  onClose: () => void;
+
+  location: Location | null;
+}
+
```

Comment:
I .. think? this is all relatively straight forward. Pretty simple modal with the `useDraftLocation` hook pretty much controlling all the things.